src/services/search/listingSearch.ts

```diff
@@ -128,11 +191,88 @@ export const fetchListings = async ({
     path: "listings/search",
     query,
     options,
+    defaultSort: defaultUserSort,
+    defaultPagination: defaultUserPagination,
+  })
+
+  return sanitizeListingResponse(response)
+}
+
+export const fetchDealerListings = async ({
+  dealerId,
+  query = {},
+  options = {},
+}: {
+  dealerId: number
+  query?: ListingQueryParams
+  options?: ApiCallOptions
+}): Promise<Paginated<SearchListing>> => {
+  const response = await searchForListings({
+    path: `dealers/${dealerId}/listings/search`,
+    query,
+    options: {
+      isAuthorizedRequest: true,
+      ...options,
+    },
+    defaultSort: defaultDealerSort,
+    defaultPagination: defaultDealerPagination,
   })
 
   return sanitizeListingResponse(response)
 }
 
+const toCamelCase = (snakeCase) => {
```

Comment:
Not sure if we should move that to lib code as someone might be more likely to look there in case the functionality is required somewhere else